tests/rgbpp/spore/launch/1-prepare-cluster.ts

```diff
@@ -0,0 +1,84 @@
+  SECP256K1_WITNESS_LOCK_SIZE,
+  append0x,
+  buildRgbppLockArgs,
+  calculateRgbppClusterCellCapacity,
+  calculateTransactionFee,
+  genRgbppLockScript,
+  getSecp256k1CellDep,
+} from 'rgbpp/ckb';
+import { ckbAddress, isMainnet, collector, CKB_PRIVATE_KEY } from '../../env';
+import { CLUSTER_DATA } from './0-cluster-info';
+import { readStepLog } from '../../shared/utils';
+
+const prepareClusterCell = async ({ outIndex, btcTxId }: { outIndex: number; btcTxId: string }) => {
+  const { retry } = await import('zx');
+  await retry(20, '10s', async () => {
+    const masterLock = addressToScript(ckbAddress);
+    console.log('ckb address: ', ckbAddress);
+
+    // The capacity required to launch cells is determined by the token info cell capacity, and transaction fee.
+    const clusterCellCapacity = calculateRgbppClusterCellCapacity(CLUSTER_DATA);
+
+    let emptyCells = await collector.getCells({
+      lock: masterLock,
+    });
+    if (!emptyCells || emptyCells.length === 0) {
+      throw new NoLiveCellError('The address has no empty cells');
+    }
+    emptyCells = emptyCells.filter((cell) => !cell.output.type);
+
+    const txFee = MAX_FEE;
+    const { inputs, sumInputsCapacity } = collector.collectInputs(emptyCells, clusterCellCapacity, txFee);
+
+    const outputs: CKBComponents.CellOutput[] = [
+      {
+        lock: genRgbppLockScript(buildRgbppLockArgs(outIndex, btcTxId), isMainnet),
+        capacity: append0x(clusterCellCapacity.toString(16)),
+      },
+    ];
+    let changeCapacity = sumInputsCapacity - clusterCellCapacity;
+    outputs.push({
+      lock: masterLock,
+      capacity: append0x(changeCapacity.toString(16)),
+    });
+    const outputsData = ['0x', '0x'];
+
+    const emptyWitness = { lock: '', inputType: '', outputType: '' };
+    const witnesses = inputs.map((_, index) => (index === 0 ? emptyWitness : '0x'));
+
+    const cellDeps = [getSecp256k1CellDep(isMainnet)];
+
+    const unsignedTx = {
+      version: '0x0',
+      cellDeps,
+      headerDeps: [],
+      inputs,
+      outputs,
+      outputsData,
+      witnesses,
+    };
+
+    const txSize = getTransactionSize(unsignedTx) + SECP256K1_WITNESS_LOCK_SIZE;
+    const estimatedTxFee = calculateTransactionFee(txSize);
+    changeCapacity -= estimatedTxFee;
+    unsignedTx.outputs[unsignedTx.outputs.length - 1].capacity = append0x(changeCapacity.toString(16));
+
+    const signedTx = collector.getCkb().signTransaction(CKB_PRIVATE_KEY)(unsignedTx);
+    const txHash = await collector.getCkb().rpc.sendTransaction(signedTx, 'passthrough');
+
+    console.info(`Cluster cell has been prepared and the tx hash ${txHash}`);
+    console.info(`explorer: https://pudge.explorer.nervos.org/transaction/${txHash}`);
+  });
+};
+
+// Please use your real BTC UTXO information on the BTC Testnet
+prepareClusterCell({
+  // outIndex: 3,
+  // btcTxId: 'aee4e8e3aa95e9e9ab1f0520714031d92d3263262099dcc7f7d64e62fa2fcb44',
+  outIndex: readStepLog('0').index,
```

Comment:
The Spore log files are indistinguishable from xUDT log files.

Reply:
Yes, a tx_id coverage log file will be re-created before the spore test starts. If it is modified to a parallel test, it needs to be renamed.